pkg/instancetype/instancetype.go

```diff
@@ -759,36 +840,73 @@ func (m *InstancetypeMethods) inferDefaultsFromDataVolumeSourceRef(sourceRef *v1
 }
 
 func applyCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
-	if vmiSpec.Domain.CPU != nil {
-		return Conflicts{field.Child("domain", "cpu")}
+	var cpuConflicts Conflicts
+
+	if vmiSpec.Domain.CPU == nil {
+		vmiSpec.Domain.CPU = &virtv1.CPU{}
 	}
 
 	if _, hasCPURequests := vmiSpec.Domain.Resources.Requests[k8sv1.ResourceCPU]; hasCPURequests {
-		return Conflicts{field.Child("domain", "resources", "requests", string(k8sv1.ResourceCPU))}
+		cpuConflicts = append(cpuConflicts, field.Child("domain", "resources", "requests", string(k8sv1.ResourceCPU)))
 	}
 
 	if _, hasCPULimits := vmiSpec.Domain.Resources.Limits[k8sv1.ResourceCPU]; hasCPULimits {
-		return Conflicts{field.Child("domain", "resources", "limits", string(k8sv1.ResourceCPU))}
+		cpuConflicts = append(cpuConflicts, field.Child("domain", "resources", "limits", string(k8sv1.ResourceCPU)))
 	}
 
-	vmiSpec.Domain.CPU = &virtv1.CPU{
-		Sockets:               uint32(1),
-		Cores:                 uint32(1),
-		Threads:               uint32(1),
-		Model:                 instancetypeSpec.CPU.Model,
-		DedicatedCPUPlacement: instancetypeSpec.CPU.DedicatedCPUPlacement,
-		IsolateEmulatorThread: instancetypeSpec.CPU.IsolateEmulatorThread,
-		NUMA:                  instancetypeSpec.CPU.NUMA.DeepCopy(),
-		Realtime:              instancetypeSpec.CPU.Realtime.DeepCopy(),
+	if vmiSpec.Domain.CPU.Sockets != 0 {
+		cpuConflicts = append(cpuConflicts, field.Child("domain", "cpu", "sockets"))
 	}
 
-	// Default to PreferSockets when a PreferredCPUTopology isn't provided
-	preferredTopology := instancetypev1beta1.PreferSockets
-	if preferenceSpec != nil && preferenceSpec.CPU != nil && preferenceSpec.CPU.PreferredCPUTopology != nil {
-		preferredTopology = *preferenceSpec.CPU.PreferredCPUTopology
+	if vmiSpec.Domain.CPU.Cores != 0 {
+		cpuConflicts = append(cpuConflicts, field.Child("domain", "cpu", "cores"))
+	}
+
+	if vmiSpec.Domain.CPU.Threads != 0 {
+		cpuConflicts = append(cpuConflicts, field.Child("domain", "cpu", "threads"))
+	}
+
+	if vmiSpec.Domain.CPU.Model != "" && instancetypeSpec.CPU.Model != nil {
+		cpuConflicts = append(cpuConflicts, field.Child("domain", "cpu", "model"))
+	} else if vmiSpec.Domain.CPU.Model == "" && instancetypeSpec.CPU.Model != nil {
```

Comment:
Does it matter the order of the checks for `!= nil` in go? Not quite sure, otherwise you should flip them